src/Shared/TagWriter/TagWriter.cs

```diff
@@ -80,7 +79,7 @@ internal abstract class TagWriter<TTagState, TArrayState>
             default:
                 try
                 {
-                    var stringValue = TruncateString(Convert.ToString(tag.Value, CultureInfo.InvariantCulture), tagValueMaxLength);
+                    var stringValue = TruncateString(Convert.ToString(tag.Value/*TODO: , CultureInfo.InvariantCulture*/), tagValueMaxLength);
```

Comment:
Curious, why is this changed to TODO?

Reply:
The existing `TagTransformer<T>` we are migrating from doesn't use `InvariantCulture` today: https://github.com/open-telemetry/opentelemetry-dotnet/blob/eace49c2fb64656edf2de2ab9e8848596a3a3e40/src/Shared/TagTransformer.cs#L75

There is a unit test in OtlpExporter which is adding a `DateTime.UtcNow` as a tag value. With this change to use `InvariantCulture` that test fails because the format changes.

I discussed a bit with @alanwest. We feel this is a bug. We should be using `InvariantCulture` in the transformation code. But it is also a breaking change. So the idea was let's do this migration to `TagWriter<T>` without changing that behavior. Then we can do a dedicated PR/fix for that after the migration so it is independent/clearly documented.